### PR TITLE
Formal click event for Icon

### PR DESCRIPTION
**Pull request by anmoljagetia**
addressing #384.

### CODE REVIEW

**Comment by GordonSmith** [comment]
Can you take a look at: https://github.com/hpcc-systems/HPCC-Platform/wiki/Git-commit-message-guidelines  Which is a rough guide for commit messages.

**Comment by GordonSmith** [comment]
I rebased this pull request onto master (without the merge) here:  https://github.com/GordonSmith/Visualization/commits/iconClick if you can take that and ammend the commit with a better msg that would be great.

File: src/common/Icon.js [hunk 1]
```diff
@@ -44,6 +44,11 @@
             .render()
         ;
         this._tooltipElement = element.append("title");
+        this.element().on("click", this.click);
```
Comment (GordonSmith) [inline src/common/Icon.js]: I think ```element.on("click", this.click);``` is equivalent.

**Comment by mlzummo** [comment]
are we going with onEvent(event,callback) ?

**Comment by anmoljagetia** [comment]
So should I change the messsage to GH384 - Adds a formal click event to icon, or anything better?

**Comment by mlzummo** [comment]
Do you have a chat client or can you got in gitter.im?

**Comment by mlzummo** [comment]
"Adds Formal Click Event #384"

**Comment by GordonSmith** [comment]
@mlzummo The click handler here is similar to all the other click handlers, (but we will have a discussion about "publishing" events at some point in the future).

@anmoljagetia - If you can change your commit message to adhere to the recommended format I will pull:
```
GH-384 Icon Click Event

Adds a click event to the icon.

Fixes GH-384

Signed-off-by: Anmol Jagetia anmoljagetia@lexisnexis.com
```

Note: you get the signature via git with:  ```git commit -s```

File: src/common/Icon.js [hunk 2]
```diff
@@ -44,6 +44,11 @@
             .render()
         ;
         this._tooltipElement = element.append("title");
+        element().on("click", this.click);
```
Comment (GordonSmith) [inline src/common/Icon.js]: should be ```element.on("click", this.click);```

**Comment by GordonSmith** [comment]
Also when issuing the PR - it would be very helpful if you could add a link to the test page, in this case:  http://rawgit.com/anmoljagetia/Visualization/iconClick/demos/dermatology.html?src/common/Icon

**Comment by GordonSmith** [comment]
Ok - looks good!

**Pull request merged**